ComponentKit/Utilities/CKComponentAction.h

```diff
@@ -125,21 +132,39 @@ class CKTypedComponentAction : public CKTypedComponentActionBase {
 
   /** We support promotion from actions that take no arguments. */
   template <typename... Ts>
-  CKTypedComponentAction<Ts...>(const CKTypedComponentAction<> &action) noexcept : CKTypedComponentActionBase(action) { };
+  CKTypedComponentAction<Ts...>(const CKTypedComponentAction<> &action) noexcept : CKTypedComponentActionBase(action) {
+    if (_variant == CKTypedComponentActionVariant::Block) {
```

Comment:
It sucks checking the variant here. It'd be great if we could bump this out. Perhaps all this would move into a templated function in the internal header that this just calls to keep the inline bloat here to a minimum.